src/zcbor_common.c

```diff
@@ -279,3 +279,109 @@ int zcbor_entry_function(const uint8_t *payload, size_t payload_len,
 	}
 	return ZCBOR_SUCCESS;
 }
+
+
+/* Float16: */
+#define F16_SIGN_OFFS 15 /* Bit offset of the sign bit. */
+#define F16_EXPO_OFFS 10 /* Bit offset of the exponent. */
+#define F16_EXPO_MSK 0x1F /* Bitmask for the exponent (right shifted by F16_EXPO_OFFS). */
+#define F16_MANTISSA_MSK 0x3FF /* Bitmask for the mantissa. */
+#define F16_MAX 65520 /* Lowest float32 value that rounds up to float16 infinity.
+		       * (65519.996 rounds to 65504) */
+#define F16_MIN_EXPO 24 /* Negative exponent of the non-zero float16 value closest to 0 (2^-24) */
+#define F16_MIN (1.0f / (1 << F16_MIN_EXPO)) /* The non-zero float16 value closest to 0 (2^-24) */
+#define F16_MIN_NORM (1.0f / (1 << 14)) /* The normalized float16 value closest to 0 (2^-14) */
+#define F16_BIAS 15 /* The exponent bias of normalized float16 values. */
+
+/* Float32: */
+#define F32_SIGN_OFFS 31 /* Bit offset of the sign bit. */
+#define F32_EXPO_OFFS 23 /* Bit offset of the exponent. */
+#define F32_EXPO_MSK 0xFF /* Bitmask for the exponent (right shifted by F32_EXPO_OFFS). */
+#define F32_MANTISSA_MSK 0x7FFFFF /* Bitmask for the mantissa. */
+#define F32_BIAS 127 /* The exponent bias of normalized float32 values. */
+
+/* Rounding: */
+#define SUBNORM_ROUND_MSK (F32_MANTISSA_MSK | (1 << F32_EXPO_OFFS)) /* mantissa + lsb of expo for
+								     * tiebreak. */
+#define SUBNORM_ROUND_BIT_MSK (1 << (F32_EXPO_OFFS - 1)) /* msb of mantissa (0x400000) */
+#define NORM_ROUND_MSK (F32_MANTISSA_MSK >> (F16_EXPO_OFFS - 1)) /* excess mantissa when going from
+								  * float32 to float16 + 1 extra bit
+								  * for tiebreak. */
+#define NORM_ROUND_BIT_MSK (1 << (F32_EXPO_OFFS - F16_EXPO_OFFS - 1)) /* bit 12 (0x1000) */
+
+
+float zcbor_float16_to_32(uint16_t input)
+{
+	uint32_t sign = input >> F16_SIGN_OFFS;
+	uint32_t expo = (input >> F16_EXPO_OFFS) & F16_EXPO_MSK;
+	uint32_t mantissa = input & F16_MANTISSA_MSK;
+
+	if ((expo == 0) && (mantissa != 0)) {
+		/* Subnormal float16 - convert to normalized float32 */
+		return ((float)mantissa * F16_MIN) * (sign ? -1 : 1);
+	} else {
```

Comment:
little nit: `else` is actually unnecessary

Reply:
I will keep it for now, as I think it makes it a little clearer. Also, I want this code to be moved with as few changes as possible.